packages/transformer/src/IModelTransformer.ts

```diff
@@ -2234,6 +2239,11 @@ export class IModelTransformer extends IModelExportHandler {
     const sourceVersion = `${this.sourceDb.changeset.id};${this.sourceDb.changeset.index}`;
     const targetVersion = `${this.targetDb.changeset.id};${this.targetDb.changeset.index}`;
 
+    if (saveReverseVersion) {
+      this._targetScopeProvenanceProps.jsonProperties.reverseSyncVersion =
+        targetVersion;
+    }
```

Comment:
Current implementation works, but I think there might be maintainability/functionality issues in future. 
My concerns are that: 
- as documentation says `saveReverseVersion` is needed only for processAll transformation (I think should say 'forward transformation' instead). In all other scenarios, reverseVersion is saved anyway. With current implementation, synch version is set to invalid value for second for reverse synch. This is not noticible as it  is fixed right away in code below: 
![image](https://github.com/user-attachments/assets/a05b27cb-a533-4916-8529-07aacbcbd245)

If that code is changed, value might get corrupted for processChanges in reverse direction.

Suggest to save reverse synch only for scenarios when it might be empty, and to set it to correct value only: 
![image](https://github.com/user-attachments/assets/ebd4f79a-23d3-4cbf-aef3-cfd848e4bc0b)

Reply:
Definitely agree with you that this function is getting a bit out of control.
Wehn you say "With current implementation, synch version is set to invalid value for second for reverse synch. This is not noticible as it is fixed right away in code below:" could you elaborate? In what scenario would we pass saveReverseVersion during a reverseSync?

Reply:
Clarified offline